scipy/spatial/qhull.pyx

```diff
@@ -2406,9 +2523,183 @@ class Voronoi(_QhullUser):
 
         _QhullUser._update(self, qhull)
 
+    def add_points(self, points, restart=False):
+        _QhullUser._add_points(self, points, restart)
+
     @property
     def ridge_dict(self):
         if self._ridge_dict is None:
             self._ridge_dict = dict(zip(map(tuple, self.ridge_points.tolist()),
                                         self.ridge_vertices))
         return self._ridge_dict
+
+#------------------------------------------------------------------------------
+# Halfspace Intersection
+#------------------------------------------------------------------------------
+
+class HalfspaceIntersection(_QhullUser):
+    """
+    HalfspaceIntersection(halfspaces, feasible_point, incremental=False, qhull_options=None)
+
+    Halfspace intersections in N dimensions. This reproduces the "qhalf"
+    functionality of Qhull.
+
+    .. versionadded:: 0.19.0
+
+    Parameters
+    ----------
+    halfspaces : ndarray of floats, shape (nineq, ndim+1)
+        Stacked Inequalities of the form Ax + b <= 0 in format [A; b]
+    feasible_point: ndarray of floats, shape (1, ndim)
```

Comment:
Qhull calls this the "interior point" --- should it use the same terminology?
Also, should it be computed automatically, if the user does not provide it?